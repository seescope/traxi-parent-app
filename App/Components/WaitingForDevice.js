import React from 'react';
import { Text, View, Dimensions, ActivityIndicator } from 'react-native';

import HeaderText from '../Components/HeaderText';
import KidAvatar from '../Components/KidAvatar';
import Spacing from '../Components/Spacing';
import Button from '../Components/Button';
import { WHITE, TRANSPARENT } from '../Constants/Colours';

const { width } = Dimensions.get('window');

const style = {
  container: {
    alignItems: 'center',
  },
  bodyText: {
    fontFamily: 'Raleway-Regular',
    fontSize: 16,
    color: WHITE,
    textAlign: 'center',
    backgroundColor: TRANSPARENT,
  },
  buttonContainer: {
    width: width - 64,
    alignItems: 'center',
  },
};

const WaitingForDevice = ({ kidName, avatarURL, deviceType }) => (
  <View style={style.container}>
    <HeaderText>
      Waiting for traxi to {'\n'}
      start on {kidName}'s {deviceType}
    </HeaderText>

    <Spacing height={32} />

    <KidAvatar
      size={166}
      avatarURL={avatarURL}
      state="neutral"
    />

    <Spacing height={32} />

    <ActivityIndicator size="large" />

    <Spacing height={32} />

    <Text style={style.bodyText}>
      Waiting...
    </Text>

    <Spacing />

    <View style={style.buttonContainer}>
      <Button primary={false} onPress={() => {}}>
        I need help
      </Button>
    </View>
  </View>
);

WaitingForDevice.propTypes = {
  kidName: React.PropTypes.string.isRequired,
  avatarURL: React.PropTypes.string.isRequired,
  deviceType: React.PropTypes.string.isRequired,
};

export default WaitingForDevice;
