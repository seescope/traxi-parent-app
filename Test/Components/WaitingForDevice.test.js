import React from 'react';
import renderer from 'react-test-renderer';
import WaitingForDevice from '../../App/Components/WaitingForDevice';

it('renders', () => {
  const tree = renderer.create(<WaitingForDevice
    kidName="Jeff"
    avatarURL="yep"
    deviceType="Android"
  />).toJSON();
  expect(tree).toMatchSnapshot();
});
