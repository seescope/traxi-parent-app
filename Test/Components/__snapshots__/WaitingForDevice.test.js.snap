// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#545454",
        "fontFamily": "Raleway-ExtraBold",
        "fontSize": 26,
        "marginBottom": -2,
        "textAlign": "center",
      }
    }
  >
    [missing "en.waitingForDevice.header" translation]
  </Text>
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <Image
    animation={undefined}
    duration={undefined}
    iterationCount={undefined}
    source={
      Object {
        "uri": "yep",
      }
    }
    style={
      Object {
        "borderColor": "#FFF",
        "borderRadius": 83,
        "borderWidth": 4.742857142857143,
        "height": 166,
        "marginBottom": 16.6,
        "shadowColor": "rgba(0,0,0,0.5)",
        "shadowOffset": Object {
          "height": 2,
        },
        "shadowRadius": 5,
        "width": 166,
      }
    }
  />
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <ActivityIndicator
    animating={true}
    color="white"
    hidesWhenStopped={true}
    size="large"
  />
  <View
    style={
      Object {
        "height": 32,
      }
    }
  />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "transparent",
        "color": "#FFF",
        "fontFamily": "Raleway-Regular",
        "fontSize": 16,
        "textAlign": "center",
      }
    }
  >
    [missing "en.waitingForDevice.waiting" translation]
  </Text>
  <View
    style={
      Object {
        "height": 24,
      }
    }
  />
  <View
    style={
      Object {
        "alignItems": "center",
        "width": 686,
      }
    }
  >
    <View
      accessibilityComponentType={undefined}
      accessibilityLabel={undefined}
      accessibilityTraits={undefined}
      accessible={true}
      hitSlop={undefined}
      isTVSelectable={true}
      onLayout={undefined}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "transparent",
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 16,
          "opacity": 1,
        }
      }
      testID={undefined}
      tvParallaxProperties={undefined}
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "color": "#448AFF",
            "fontFamily": "Raleway-Regular",
            "fontSize": 14,
          }
        }
      >
        [missing "en.general.needHelp" translation]
         
      </Text>
    </View>
  </View>
</View>
`;
